python/lsst/ts/genericcamera/genericcameracsc.py

```diff
@@ -705,7 +626,164 @@ async def run_auto_exposure_loop(self, min_exp_time, max_exp_time, configuration
         self.isAutoExposure = False
         self.log.info("autoExposure_loop - End")
 
-    async def establish_exposure_background(self, exposure):
+    async def run_auto_exposure(self, min_exp_time, max_exp_time, configuration):
+        """Take auto exposures until `runAutoExposureTask` is False.
+
+        Parameters
+        ----------
+        min_exp_time: `float`
+            The minimum exposure time to use.
+        max_exp_time: `float`
+            The maximum exposure time to use.
+        configuration: `dict`
+            A dict containing additional configuration parameters.
+        """
+        while self.runAutoExposureTask:
+            timestamp = time.time()
+            image_name = self.fileNameFormat.format(
+                timestamp=int(timestamp),
+                index=configuration["image_index"],
+                total=configuration["images_in_sequence"],
+            )
+
+            self.evt_startTakeImage.put()
+            await self.camera.startTakeImage(
+                configuration["exposure_time"],
+                configuration["shutter"],
+                configuration["sensors"],
+                configuration["keyValueMap"],
+                configuration["obsNote"],
+            )
+            exposure = await self.determine_auto_exposure_time_and_get_exposure(
```

Comment:
How about `get_auto_exposure` for the name of this method?

Reply:
Much better yes. Done.